test/e2e/single_httproute_test.go

```diff
@@ -115,7 +115,7 @@ func setupSingleHTTPRouteEnv(ctx context.Context, t *testing.T, config *envconf.
 	err = wait.For(
 		waitCondition.ResourceMatch(
 			resourcesMap[HTTP_ROUTE_KEY],
-			getMatchHTTPRouteFetcher(t, FIRST_CANARY_ROUTE_WEIGHT),
+			getMatchHTTPRouteWithLabelFetcher(t, FIRST_CANARY_ROUTE_WEIGHT, false),
```

Comment:
Maybe leave the http route test as is?

And then create a brand new test for labels? Is the test behavior unchanged in all other ways?